docs/oss-syntax.md

```diff
@@ -0,0 +1,348 @@
+##  OSS Syntax
+
+ 
+Rule format is inspired by CSS, but extended to support nesting and chaining and tweaked to deal with identifiers containing “.”s. 
+
+**CSS**
+```css
+  div#toc a {
+      text-decoration:underline;
+  }
+```
+
+
+**OSS**
+ ```css
+     class=User field=email {
+         cueTip:”Your email”;
+     }
+
+```
+
+* **CSS**: if an `a` tag appears in a `div` named toc, set the property `text-decoration` to `underline`
+* **OSS**: if a field named `email` appears in a class named `User`, set the property `cueTip` to `Your email address`
+
+
+### Basic syntax
+
+#### Nesting
+
+These are equivalent
+
+```css
+class=User field=firstName editing { visible:true; }
+class=User field=lastName  editing { visible:true; }
+class=User field=email editing     { visible:true; }
+
+```
+
+or 
+
+```css
+class=User editing {
+    field=firstName { visible:true; }
+    field=lastName  { visible:true; }
+    field=email     { visible:true; }
+}
+```
+
+or 
+
+```css
+class=User editing field=(firstName, lastName) {
+    visible:true; 
+}
+```
+
+
+#### Values
+
+Values can be Boolean, Integer, String, List, Map, and Expressions 
+
+```javascript
+field=password {
+    // boolean
+    editable:true;
+    
+    //String literal
+    cueTip:"This is required";
+    
+    // List of strings
+    trait:required, secret;
```

Comment:
are the strings only comma delineated? If so is there a way to escape a comma if I have a comma in my values?  I may have strings with commas in them that I want to put in a list.